src/consensus/versionbits.h

```diff
@@ -0,0 +1,102 @@
+// Copyright (c) 2009-2015 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#ifndef BITCOIN_VERSIONBITS_H
+#define BITCOIN_VERSIONBITS_H
+
+#include <map>
+#include <set>
+#include <stdexcept>
+#include <stdint.h>
+
+namespace Consensus
+{
+
+const int           NO_RULE      = -1;
+
+namespace VersionBits
+{
+
+const int           VERSION_HIGH_BITS   = 0x20000000;
+const int           VERSION_BITS_MASK   = 0x1fffffff;
+const char          MIN_BIT             = 0;
+const char          MAX_BIT             = 28;
+
+enum RuleState { UNDEFINED, DEFINED, LOCKED_IN, ACTIVE, FAILED };
+
+const char* GetRuleStateText(int ruleState, bool bUseCaps = false);
+
+bool UsesVersionBits(int nVersion);
+
+class SoftFork
```

Comment:
Can you move this class (ideally as a struct with public attributes instead of getters) to consensus/params.h ? 
It will be eventually necessary to expose the versionbits functionality in libconsensus.